.github/workflows/build-and-push.yml

```diff
@@ -0,0 +1,38 @@
+on:
+    push:
+        branches:
+            - master
+    pull_request:
+        branches:
+            - "*"
+
+name: Build and Push Image
+
+jobs:
+    test:
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v4
+
+            - name: Login to Docker Hub
+              uses: docker/login-action@v3
+              with:
+                username: ${{ secrets.DOCKERHUB_USERNAME }}
+                password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+            - name: Make shared env
+              uses: ./.github/actions/make-shared-env
+
+            - name: Build app
+              env:
+                NODE_ENV: production
+              run: src/gradlew -p src :app:bootDistTar
+
+            - name: Build docker image
+              run: docker build -f ./Dockerfile --tag "${CI_ORG}/hint:${CI_SHA:0:7}" src
+
+            - name: Push image
+              run: |
+                docker tag "${CI_ORG}/hint:${CI_SHA:0:7}" "${CI_ORG}/hint:${CI_BRANCH}"
+                docker push "${CI_ORG}/hint:${CI_SHA:0:7}"
+                docker push "${CI_ORG}/hint:${CI_BRANCH}"
```

Comment:
Can we push these to github container registry too? See https://github.com/mrc-ide/hintr/blob/master/.github/workflows/docker.yaml

And fairly pedantic but can we name this .yml something to do with docker? `docker.yml` or `build-and-push-docker`?